### DISCUSSION TITLE
Can repositories plugin not be used within markdown template?

### DISCUSSION THREAD

**Discussion by Andre601** [lowlighter/metrics#1194] [Q&A] [answered]
Tried to use the `repositories_starred` and `repositories_random` settings within my Profile readme using the `{%- await embed(...) %}` format, but when the action is running are empty SVGs returned in the final Readme file.

Is the plugin not supported for the `markdown` template?
I did enable it in the workflow file using `plugin_repositories: yes`

Profile readme: https://github.com/Andre601/Andre601

**Comment by lowlighter** [comment, marked as answer]
You also need to tell the `embed` function that you want to use the `repositories` plugin by passing the flag too in its option like below:

```diff
- {%- await embed(`stats-starred-repo`, {repositories_starred: 3}) %}
+ {%- await embed(`stats-starred-repo`, {repositories: true, repositories_starred: 3}) %}

- {%- await embed(`stats-random-repo`, {repositories_random: 3}) %}
+ {%- await embed(`stats-random-repo`, {repositories: true, repositories_random: 3}) %}
```

Should do the trick 👍

**Reply by Andre601** [reply]
Weird, I could've sworn they didn't need to be enabled if you do so in the workflow file...

~~Maybe make the action display a text in the SVG like "Add `<plugin>: yes` to this block to enable it" when it isn't present?~~
~~Like some info that they have to explicitly toggle it.~~

Seems to be covered by #1196

**Reply by lowlighter** [reply]
Yeah, the reason why is because you can use `embed` multiple times within the same markdown template, but you may want to display differents plugins for each render so you need to tell the function what exactly should be displayed for a given image

I don't remember whether there are limitations behind the fact that plugins must still be enabled at workflow level (for tokens it's mandatory, since it wouldn't be acceptable to just put raw token in a template 😅) but possibly we could also add an "auto-enable" mode with #1196 to enable on the fly plugin depending on which options are passed
